### PR TITLE
Added Go subcommand: `job kill`.

### PR DESCRIPTION
**Pull request by rgo3**
https://jira.mesosphere.com/browse/DCOS_OSS-4133

### CODE REVIEW

File: pkg/cmd/job/job_kill.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := metronomeClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			var deadpool []string
+			if len(args) == 1 && all == true {
+				runs, err := client.Runs(args[0])
```
Comment (bamarni) [inline pkg/cmd/job/job_kill.go]: How the runs from here are different from the ones in `Job.ActiveRuns`?

Comment (rgo3) [reply pkg/cmd/job/job_kill.go]: We could get them through that field too but I figured I would use the extra endpoint the API provides for it.

File: pkg/cmd/job/job_kill.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdJobKill kills a run of a job
+func newCmdJobKill(ctx api.Context) *cobra.Command {
+	var all bool
+	cmd := &cobra.Command{
+		Use:   "kill <job-id> [OPTIONS]",
```
[the remark this answers is not in the document]
Comment (bamarni) [reply pkg/cmd/job/job_kill.go]: sounds good

File: pkg/cmd/job/job_kill.go (hunk 1 above)
Comment (bamarni) [reply pkg/cmd/job/job_kill.go]: ok, I was just curious if there were any difference between the 2

File: pkg/cmd/job/job_kill.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
+package job
+
+import (
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdJobKill kills a run of a job
```
Comment (ArmandGrillet) [inline pkg/cmd/job/job_kill.go]: Missing period.

File: pkg/cmd/job/job_kill.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				if err != nil {
+					return err
+				}
+				ctx.Logger().Infof("Run '%s' for job '%s' killed.", dead, args[0])
+			}
+
+			return nil
+		},
+	}
+	cmd.Flags().BoolVar(&all, "all", false, "kill all active runs of this job")
```
Comment (ArmandGrillet) [inline pkg/cmd/job/job_kill.go]: Would add a `the`.

File: pkg/cmd/job/job_kill.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+			} else if len(args) == 2 {
+				deadpool = append(deadpool, args[1])
+			}
+
+			for _, dead := range deadpool {
+				err = client.Kill(args[0], dead)
+				if err != nil {
+					return err
+				}
+				ctx.Logger().Infof("Run '%s' for job '%s' killed.", dead, args[0])
```
Comment (ArmandGrillet) [inline pkg/cmd/job/job_kill.go]: s/for/of

File: pkg/cmd/job/job.go [hunk 6]
```diff
@@ -25,6 +25,7 @@ func NewCommand(ctx api.Context) *cobra.Command {
 	cmd.AddCommand(
 		newCmdJobAdd(ctx),
 		newCmdJobList(ctx),
+		newCmdJobKill(ctx),
```
Comment (ArmandGrillet) [inline pkg/cmd/job/job.go]: Please move it it before `newCmdJobList`.

File: pkg/cmd/job/job.go [hunk 7]
```diff
@@ -25,6 +25,7 @@ func NewCommand(ctx api.Context) *cobra.Command {
...
 		newCmdJobUpdate(ctx),
```
Comment (ArmandGrillet) [inline pkg/cmd/job/job.go]: Please move it after `newCmdJobRun`.

Comment (rgo3) [reply pkg/cmd/job/job.go]: I have done that on the `job schedule` command which you will see on that PR for that.

File: pkg/cmd/job/job_kill.go [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := metronomeClient(ctx)
+			if err != nil {
+				return err
+			}
+
+			var deadpool []string
+			if len(args) == 1 && all == true {
```
Comment (bamarni) [inline pkg/cmd/job/job_kill.go]: Checking all is unnecessary here as there is a pre-check. I'd fine it more clear if there were an early exit path for a specific run-id and a single loop for the `--all` flag, as reading through the shared deadpool slice and the 2 loops is harder to grasp , like this for example : 
``` go
if len(args) == 2 {
	return client.Kill(args[0], args[1])
}

runs, err := client.Runs(args[0])
if err != nil {
	return err
}
for _, run := range runs {
	err = client.Kill(args[0], run.ID)
	if err != nil {
		return err
	}
}
```

**Pull request merged**